src/providers/solr-search.php

```diff
@@ -5,34 +5,97 @@
 use SearchApi\Services\Search;
 use SearchApi\Models\SearchTerm;
 use SearchApi\Models\SearchResult;
+use SearchApi\Models\SearchResultItem;
 use SearchApi\Models\SearchOptions;
 
 use SearchApi\Builders\QueryBuilder;
 use SearchApi\Builders\SolrQueryBuilder;
 
+use SearchApi\Clients;
+
 /**
- * Class SolrSearch - This is a placeholder class that will be an interface to a Solr Search Index
+ * Interface to Solr server.
  *
- * @var QueryBuilder queryBuilder Obj with implementation of QueryBuilder constructing Solr-specific query string.
- * @method SearchResult query (SearchTerm[]|null $keywords, SearchOptions|null options) Interface defined in Services\Search
+ * @var QueryBuilder $queryBuilder Obj with implementation of QueryBuilder constructing Solr-specific query string.
+ * @var string $apiUrl Base URL for the solr query API
+ * @var HttpClient $httpClient Preferred http client wrapper for making simple GET requests.
  */
 class SolrSearch implements Search {
 
   private $queryBuilder;
+  private $apiUrl;
+  private $httpClient;
 
   /**
-   * Constructor
+   * Constructor with optional params
    *
-   * @param queryBuilder QueryBuilder Should be able to inject dependency to use any solr-specific impl. of QueryBuilder
+   * @param QueryBuilder $queryBuilder Should be able to inject dependency to use any solr-specific impl. of QueryBuilder
+   * @param HttpClient $httpClient Preferred http client wrapper. Defaults to CurlHttpClient
    */
-  function __construct( QueryBuilder $queryBuilder = null ) {
+  function __construct( QueryBuilder $queryBuilder = null, Clients\HttpClient $httpClient = null, $apiUrl = null ) {
     if ( $queryBuilder ) {
       $this->queryBuilder = $queryBuilder;
     } else {
       $this->queryBuilder = new SolrQueryBuilder();
     }
+
+    if ( $httpClient ) {
+      $this->httpClient = $httpClient;
+    } else {
+      $this->httpClient = new Clients\CurlHttpClient();
+    }
+
+    // TODO: Require $apiUrl param
+    if ( $apiUrl ) {
+      $this->apiUrl = $apiUrl;
+    } else {
+      $this->apiUrl = 'http://127.0.0.1:8983/solr/gios/select'; // TODO: get this url from a config file
+    }
   }
 
+  /**
+   * Create a search result from a JSON string.
+   * returns SearchResult
+   * @param string $responseString
+   */
+  function parse_query_response( $responseString ) {
```

Comment:
You should at least have two tests that just tests `parse_query_response` by itself. One for a good response string and one with a bad (or empty) response string.